version: '3'

services:

  nginx:
    build:
      context: ./nginx
      dockerfile: Dockerfile
    container_name: nginx
    restart: always
    ports: 
      - '8080:80'
    depends_on:
      - node
      - db
    networks:
      - desafio-network

  node:
    container_name: node
    restart: always
    tty: true
    build:
      context: ./node
      dockerfile: Dockerfile
    volumes:
      - ./node:/usr/src/app
      - /usr/src/app/node_modules
    depends_on:
      db:
        condition: service_healthy
    networks:
      - desafio-network

  db: 
    image: mysql:5.7
    command: --innodb-use-native-aio=0
    container_name: db
    restart: always
    tty: true
    volumes:
      - ./mysql/volume:/var/lib/mysql
      - ./mysql:/docker-entrypoint-initdb.d
    environment:
      - MYSQL_DATABASE=nodedb
      - MYSQL_ROOT_PASSWORD=change-me
    networks:
      - desafio-network
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      interval: 10s
      timeout: 20s
      retries: 5

networks:
  desafio-network:
    driver: bridge